### PR TITLE
use Sandbox::askForAccess for all external library features

### PR DESCRIPTION
**Pull request by Be-ing**
Follow up from #3627

### CODE REVIEW

**Comment by Holzhaus** [comment]
Thanks. Is the sandbox only restricting *opening* files, or does it apply to fit listings as well? If the latter is the case, check the `findSeratoLibraries` function, we probably need to add it there, too.

**Comment by Be-ing** [comment]
QFileInfo::exists does not require asking permission.

**Comment by Be-ing** [comment]
I think `findSeratoDatabases` is okay as it is, but it would be good for someone with a Mac and Serato to confirm that.

File: src/library/banshee/bansheefeature.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ void BansheeFeature::activate() {
             qDebug() << m_databaseFile << "does not exist";
         }
 
-        if (!m_connection.open(m_databaseFile)) {
+        if (!Sandbox::askForAccess(m_databaseFile) && !m_connection.open(m_databaseFile)) {
```
Comment (uklotzde) [inline src/library/banshee/bansheefeature.cpp:82]: I guess you intended to use `||` and not `&&`?

**Review by uklotzde — CHANGES_REQUESTED** [review]

File: src/library/banshee/bansheefeature.cpp (hunk 1 above)
Comment (Be-ing) [reply src/library/banshee/bansheefeature.cpp]: Yes, thank you. This is why we do code review. :)

File: src/library/traktor/traktorfeature.cpp [hunk 2]
```diff
@@ -216,7 +216,7 @@ TreeItem* TraktorFeature::importLibrary(const QString& file) {
 
     //Parse Trakor XML file using SAX (for performance)
     QFile traktor_file(file);
-    if (!traktor_file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+    if (!Sandbox::askForAccess(file) && !traktor_file.open(QIODevice::ReadOnly | QIODevice::Text)) {
```
Comment (uklotzde) [inline src/library/traktor/traktorfeature.cpp:219]: still wrong

Comment (Be-ing) [reply src/library/traktor/traktorfeature.cpp]: Thank you for double checking. Fixed.

**Review by uklotzde — APPROVED** [review]

File: src/library/traktor/traktorfeature.cpp (hunk 2 above)
Comment (uklotzde) [reply src/library/traktor/traktorfeature.cpp]: Unrelated: QIODevice::Text seems to be wrong here. The XML parser expects binary data. Treating XML as text is not good idea.

File: src/library/rhythmbox/rhythmboxfeature.cpp [hunk 3]
```diff
@@ -150,7 +150,8 @@ TreeItem* RhythmboxFeature::importMusicCollection() {
         }
     }
 
-    if (!db.open(QIODevice::ReadOnly | QIODevice::Text)) {
+    if (!Sandbox::askForAccess(QFileInfo(db).absoluteFilePath()) ||
```
Comment (uklotzde) [inline src/library/rhythmbox/rhythmboxfeature.cpp:153]: Same here, XML is not plain text.

**Comment by Be-ing** [comment]
Merge?

**Comment by uklotzde** [comment]
LGTM

**Pull request merged**